src/XRP/reliable-submission-xrp-client.ts

```diff
@@ -48,12 +48,69 @@ export default class ReliableSubmissionXrpClient implements XrpClientDecorator {
     sendMoneyDetails: SendXrpDetails,
   ): Promise<string> {
     const { sender } = sendMoneyDetails
-    const ledgerCloseTimeMs = 4 * 1000
 
-    // Submit a transaction hash and wait for a ledger to close.
     const transactionHash = await this.decoratedClient.sendWithDetails(
       sendMoneyDetails,
     )
+    await this.awaitFinalTransactionResult(transactionHash, sender)
+
+    return transactionHash
+  }
+
+  public async getLatestValidatedLedgerSequence(
+    address: string,
+  ): Promise<number> {
+    return this.decoratedClient.getLatestValidatedLedgerSequence(address)
+  }
+
+  public async getRawTransactionStatus(
+    transactionHash: string,
+  ): Promise<RawTransactionStatus> {
+    return this.decoratedClient.getRawTransactionStatus(transactionHash)
+  }
+
+  public async accountExists(address: string): Promise<boolean> {
+    return this.decoratedClient.accountExists(address)
+  }
+
+  public async paymentHistory(address: string): Promise<Array<XrpTransaction>> {
+    return this.decoratedClient.paymentHistory(address)
+  }
+
+  public async getPayment(
+    transactionHash: string,
+  ): Promise<XrpTransaction | undefined> {
+    return this.decoratedClient.getPayment(transactionHash)
+  }
+
+  public async enableDepositAuth(
+    wallet: Wallet,
+  ): Promise<[string, TransactionStatus]> {
+    const result = await this.decoratedClient.enableDepositAuth(wallet)
+    const transactionHash = result[0]
+
+    const rawTransactionStatus = await this.awaitFinalTransactionResult(
+      transactionHash,
+      wallet,
+    )
+
+    let transactionStatus: TransactionStatus
```

Comment:
nit: mildly cleaner to write this as:
```js
if (!rawTransactionStatus.isValidated) {
  return [hash, TransactionStatus.pending]
} else {
  const transactionStatus = ...
  return [hash, transactionStatus]
}
```

That way you don't have a dangling `let`.